pkg/graveler/ref/commit_ordered_iterator.go

```diff
@@ -91,18 +90,11 @@ func (iter *OrderedCommitIterator) maybeFetch() {
 			Prefix("NOT EXISTS (").Suffix(")").
 			From("graveler_commits c2").
 			Where("c2.repository_id=graveler_commits.repository_id").
-			Where(sq.Or{
-				sq.And{
-					sq.Lt{"c2.version": graveler.CommitVersionParentSwitch},
-					sq.Expr("c2.parents[array_length(c2.parents, 1)]=graveler_commits.id")},
-				sq.And{
-					sq.GtOrEq{"c2.version": graveler.CommitVersionParentSwitch},
-					sq.Expr("c2.parents[1]=graveler_commits.id")},
-			})
+			Where("first_parent(c2.parents, c2.version)=graveler_commits.id")
 		q = q.Where(notExistsCondition)
 	}
-	q.OrderBy("id ASC")
-	q.Limit(uint64(iter.prefetchSize))
+	q = q.OrderBy("id ASC")
+	q = q.Limit(uint64(iter.prefetchSize))
```

Comment:
I am puzzled by this style.  The Squirrel documentation _and_ [examples](https://github.com/Masterminds/squirrel/blob/cd1fe0a38ba74020c727421ca469fc4a4d7f0258/select_test.go#L15) all seem to use the assigned version.  Is this style supported?

AFAICT _this should not work_: it ends up using https://github.com/lann/ps/blob/master/map.go, which is an "immutable map", and every Set returns a *new* builder without disturbing the old one.  Please let's be very sure to test this.

Reply:
D'uh, this *adds* the chaining that I wanted.  So I just explained in a long comment why this change is good.  Oops :stuck_out_tongue_winking_eye: sorry